modules/preview-react/radio/lib/StyledRadio.tsx

```diff
@@ -0,0 +1,191 @@
+      opacity: disabled && variant === 'inverse' ? '.4' : '1',
+    },
+
+    '&:hover:after': {
+      borderColor: disabled
+        ? inputColors.disabled.border
+        : variant === 'inverse'
+        ? colors.soap300
+        : colors.licorice500,
+    },
+
+    '&:focus:after': {
+      borderColor: disabled
+        ? inputColors.disabled.border
+        : variant === 'inverse'
+        ? colors.soap300
+        : colors.blueberry400,
+    },
+
+    '&:checked:after': {
+      backgroundColor: variant === 'inverse' ? themePrimary.main : colors.frenchVanilla100,
+      border: `5px solid`,
+      borderColor: variant === 'inverse' ? colors.frenchVanilla100 : themePrimary.main,
+    },
+
+    '&:focus:after, &:focus:hover:after': {
+      outline: 'transparent',
+
+      ...focusRing({
+        width: variant === 'inverse' ? 2 : 1,
+        separation: 0,
+        animate: false,
+        innerColor: variant === 'inverse' ? colors.blackPepper400 : colors.frenchVanilla100,
+        outerColor: variant === 'inverse' ? colors.frenchVanilla100 : colors.blueberry400,
+      }),
+    },
+
+    '&:focus:checked:after, &:focus:hover:checked:after': {
+      outline: 'transparent',
+      ...focusRing({
+        width: variant === 'inverse' ? 2 : 2,
+        separation: 2,
+        animate: false,
+        innerColor: variant === 'inverse' ? colors.blackPepper400 : colors.frenchVanilla100,
+        outerColor: variant === 'inverse' ? colors.frenchVanilla100 : colors.blueberry400,
+      }),
+    },
+
+    ...mouseFocusBehavior({
+      '&:focus:after': {
+        ...focusRing({
+          width: 0,
+          outerColor: variant === 'inverse' ? colors.frenchVanilla100 : themeFocusOutline,
+        }),
+      },
+      '&:focus:hover:after, &:focus:active:after': {
+        ...focusRing({
+          width: 0,
+          outerColor: variant === 'inverse' ? colors.frenchVanilla100 : themeFocusOutline,
+        }),
+      },
+      '&:focus:hover:checked:after, &:focus:active:checked:after': {
+        ...focusRing({
+          width: 0,
+          outerColor: variant === 'inverse' ? colors.frenchVanilla100 : themeFocusOutline,
+        }),
+      },
+    }),
+  })
+);
+
+const RadioInputWrapper = styled(Flex)<Pick<StyledRadioProps, 'disabled' | 'variant'>>(
+  {
+    // Hover Ripple element
+    '::before': {
+      content: "''",
+      borderRadius: borderRadius.circle,
+      height: radioHeight,
+      transition: 'box-shadow 150ms ease-out',
+      width: radioWidth,
+      pointerEvents: 'none',
+    },
+  },
+  ({variant, disabled}) => ({
+    '::before': {
+      opacity: variant === 'inverse' ? '0.4' : '1',
+    },
+    '&:hover:before': {
+      boxShadow: disabled ? undefined : `0 0 0 ${rippleRadius}px ${colors.soap200}`,
+    },
+  })
+);
+
+export interface StyledRadioProps extends ExtractProps<typeof Box, 'input'> {
+  variant?: 'inverse' | undefined;
+}
+
+export const StyledRadio = createComponent('input')({
```

Comment:
What do you think about renaming `StyledRadio` to `StyledRadioButton` for consistency (since we have a `RadioGroup.RadioButton` subcomponent)?

Reply:
that works for me

Reply:
We'll need to rename `StyledRadio` to `StyledRadioButton` in the docs as well.